utils/utils.go

```diff
@@ -75,3 +95,161 @@ func IsValidEthereumAddress(address string) bool {
 	re := regexp.MustCompile("^0x[0-9a-fA-F]{40}$")
 	return re.MatchString(address)
 }
+
+func ReadPublicUrl(url string) ([]byte, error) {
+	// allow no redirects
+	httpClient := http.Client{
+		CheckRedirect: func(req *http.Request, via []*http.Request) error {
+			return http.ErrUseLastResponse
+		},
+		Timeout: 3 * time.Second,
+	}
+	resp, err := httpClient.Get(url)
+	if err != nil {
+		return []byte{}, err
+	}
+
+	if resp.StatusCode >= 400 {
+		return []byte{}, fmt.Errorf("error fetching url: %s", resp.Status)
+	}
+
+	defer func(Body io.ReadCloser) {
+		err := Body.Close()
+		if err != nil {
+			fmt.Println("error closing url body")
+		}
+	}(resp.Body)
+
+	return io.ReadAll(resp.Body)
+}
+
+func CheckIfValidTwitterURL(twitterURL string) error {
+	// Basic validation
+	err := CheckBasicURLValidation(twitterURL)
+	if err != nil {
+		return err
+	}
+
+	// Regular expression to validate URLs
+	urlPattern := regexp.MustCompile(`^(?:https?://)?(?:www\.)?(?:twitter\.com/\w+|x\.com/\w+)(?:/?|$)`)
+
+	// Check if the URL matches the regular expression
+	if !urlPattern.MatchString(twitterURL) {
+		return ErrInvalidTwitterUrlRegex
+	}
+
+	return nil
+}
+
+func CheckBasicURLValidation(rawUrl string) error {
+	if len(rawUrl) == 0 {
+		return ErrEmptyUrl
+	}
+
+	if strings.Contains(rawUrl, "localhost") || strings.Contains(rawUrl, "127.0.0.1") {
+		return ErrUrlPointingToLocalServer
+	}
+
+	if len(rawUrl) > 1024 {
+		return ErrInvalidUrlLength
+	}
+
+	parsedURL, err := url.Parse(rawUrl)
+	if err != nil {
+		return err
+	}
+
+	// Check if the URL is valid
+	if parsedURL.Scheme != "" && parsedURL.Host != "" {
+		return nil
+	} else {
+		return ErrInvalidUrl
+	}
+}
+
+func CheckIfUrlIsValid(rawUrl string) error {
+	// Basic validation
+	err := CheckBasicURLValidation(rawUrl)
+	if err != nil {
+		return err
+	}
+
+	// Regular expression to validate URLs
+	urlPattern := regexp.MustCompile(`^(https?)://[^\s/$.?#].[^\s]*$`)
+
+	// Check if the URL matches the regular expression
+	if !urlPattern.MatchString(rawUrl) {
+		return ErrInvalidUrl
+	}
+
+	return nil
+}
+
+func IsImageURL(urlString string) error {
+	// Parse the URL
+	parsedURL, err := url.Parse(urlString)
+	if err != nil {
+		return err
+	}
+
+	// Extract the path component from the URL
+	path := parsedURL.Path
+
+	// Get the file extension
+	extension := filepath.Ext(path)
```

Comment:
tested with files like malware.sh.png, it will return png , although the file type maynot be png

Reply:
yea the mimetype check will fail - do you want any stricter check?